src/JSONLD/JSONLD_Web.jl

```diff
@@ -1,14 +1,18 @@
 struct JSONLD_Web <: JSONLD end
 
 function mainpage_url(repo::JSONLD_Web, dataname)
-    page=getpage(dataname)
-    pattern = sel"script[type=\"application/ld+json\"]"
-    jsonld_blocks = matchall(pattern, page.root)
-    if length(jsonld_blocks)==0
-        error("No JSON-LD Linked Data Found")
+    try
+        page=getpage(dataname)
+        pattern = sel"script[type=\"application/ld+json\"]"
+        jsonld_blocks = matchall(pattern, page.root)
+        if length(jsonld_blocks)==0
+            error("No JSON-LD Linked Data Found")
+        end
+        @assert length(jsonld_blocks)==1
+        script_block = text_only(first(jsonld_blocks))
+        json = JSON.parse(script_block)
+        return json, dataname
+    catch
+        throw(GeneratorError(repo))
```

Comment:
Delete the `try`/`catch` here.
The error that is thrown is more informative anyway.
We want to show the error to the user at the top level,
so that if they were expecting this to work, they know why it didn't.
Thus we do not want to hide the original exception message.


You could add an optional message field (default to `""`) to `GeneratorError` and,
change line 9, to be 
`throw(GeneratorError, repo, "No JSON-LD Linked Data Found")`